src/codecs/browser-jpeg/decoder.ts

```diff
@@ -0,0 +1,15 @@
+import { canDecode, fileToBitmap } from '../../lib/util';
+
+export const name = 'Browser JPEG Decoder';
+export const supportedExtensions = ['jpg', 'jpeg'];
+export const supportedMimeTypes = ['image/jpeg'];
+export async function decode(file: File): Promise<ImageBitmap> {
+  return fileToBitmap(file);
+}
+
+// tslint:disable-next-line:max-line-length It’s a data URL. Whatcha gonna do?
+const jpegFile = 'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/2wBDAQMDAwQDBAgEBAgQCwkLEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBD/wAARCAABAAEDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAn/xAAUEAEAAAAAAAAAAAAAAAAAAAAA/8QAFAEBAAAAAAAAAAAAAAAAAAAAAP/EABQRAQAAAAAAAAAAAAAAAAAAAAD/2gAMAwEAAhEDEQA/AJVAA//Z';
+
+export function isSupported(): Promise<boolean> {
+  return canDecode(jpegFile);
+}
```

Comment:
Part of me likes the robustness here, but I think it's overkill. If our target browsers are all capable of decoding a particular format, I don't think we need a feature test for it.

I haven't looked at the rest of the PR yet, so this may be nonsense, but I figured the pattern would be:

1. Look at the first n bytes of the file. (I think this is the best method rather than looking at type or extension. It's certainly what the browser does).
1. If it's a type that isn't universally supported by our target browsers, then:
    1. If this browser doesn't support decoding natively, then return the result of decoding it using the relevant WASM decoder.
1. Return the result of giving it to the native decoder.

Reply:
In terms of the sniffing, all WebPs start with bytes `RIFF` followed by 4 bytes that can be anything, then bytes `WEBPVP8 ` (including the space at the end).

This is what Chrome uses: https://cs.chromium.org/chromium/src/net/base/mime_sniffer.cc?gsn=mask&l=155

Reply:
That sounds like a good idea - I don't think Firefox even reports a mimetype for webp files.

Reply:
Seems like this logic should sit in `fileToBitmap`.